core/src/components/datetime/utils/data.ts

```diff
@@ -254,7 +254,22 @@ export const generateTime = (
  * Given DatetimeParts, generate the previous,
  * current, and and next months.
  */
-export const generateMonths = (refParts: DatetimeParts): DatetimeParts[] => {
+export const generateMonths = (refParts: DatetimeParts, forcedDate: DatetimeParts | null = null): DatetimeParts[] => {
```

Comment:
This might be better as `forcedDate?: DatetimeParts`. We call `generateMonths(workingParts)` in `renderCombinedDatePickerColumn`, so `forcedDate` is technically `undefined` here.